api/src/main/java/org/apache/iceberg/expressions/UnboundPredicate.java

```diff
@@ -125,13 +132,26 @@ public Expression bind(Types.StructType struct, boolean caseSensitive) {
         case LT_EQ:
         case EQ:
           return Expressions.alwaysFalse();
-//        case IN:
-//          break;
-//        case NOT_IN:
-//          break;
+        case IN:
+        case NOT_IN:
+          break;
       }
     }
-    return new BoundPredicate<>(op(), new BoundReference<>(field.fieldId(),
-        schema.accessorForField(field.fieldId())), lit);
+
+    switch (op()) {
+      case IN:
+      case NOT_IN:
+        @SuppressWarnings("unchecked")
+        Set<Literal<T>> lits = literalSet()
+                .stream()
+                .map(l -> (Literal<T>) l.to(field.type()))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+        return new BoundPredicate<>(op(), new BoundReference<>(field.fieldId(),
```

Comment:
`aboveMax` and `belowMin` should be handled in this method. Either one indicates that no values are equal to the original literal. For both `in` and `notIn`, I think those values can be filtered out of the set.

Reply:
👍